File: 0328-odd-even-linked-list/0328-odd-even-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode oddEvenList(ListNode head) {
        ArrayList<Integer> one = new ArrayList<>();
        ArrayList<Integer> two = new ArrayList<>();
        ArrayList<Integer> ans = new ArrayList<>();
        ListNode temp = head;
        while(temp!=null){
            ans.add(temp.val);
            temp=temp.next;

        }
        int i=0;
        while(i<ans.size()){
            int t = ans.get(i);
            one.add(t);
            i+=2;
        }

        int j=1; 
        while(j<ans.size()){
            int t = ans.get(j);
            two.add(t);
            j+=2;
        }

        ListNode dummy = new ListNode(0);
        ListNode dummy1 = dummy;
        for(int v : one){
            dummy1.next = new ListNode(v);
            dummy1=dummy1.next;
        }
        for(int v : two){
            dummy1.next = new ListNode(v);
            dummy1=dummy1.next;
        }
        return dummy.next;
        


    }
}

File: 0328-odd-even-linked-list/0328-odd-even-linked-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def oddEvenList(self, head):
        """
        :type head: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        one = []
        if head == None or head.next == None:
            return head
        slow = head
        while slow!=None:
            one.append(slow.val)
            if slow.next != None and  slow.next.next != None :
                slow=slow.next.next
            else:
                break


            
        two = []
        slow = head.next
        while slow!=None:
            two.append(slow.val)
            if slow.next != None and slow.next.next != None:
                slow = slow.next.next
            else:
                break
            

        final = one + two
        dummy1 = ListNode(0)
        dummy2 = dummy1
        for num in final:
            dummy2.next = ListNode(num)
            dummy2 = dummy2.next
        return dummy1.next
        